### ISSUE TITLE
pre-commit githook --exclude is ignored

### ISSUE THREAD

**Issue by martinmose** [nicklockwood/SwiftFormat#1073]
Hello,

I have encountered a problem where `--exclude` is ignored from the `.swiftformat` file. 
The issue only occur when swiftformat is triggered from a githook file.

My pre-commit githook look like this:

```bash
#!/usr/bin/env zsh

if test -d "$HOME/.mint/bin"; then
 PATH="$HOME/.mint/bin:${PATH}"
fi

export PATH

if test ! $(which swiftformat); then
  1>&2 echo "warning: SwiftFormat not installed, run setup.sh (or download from https://github.com/nicklockwood/SwiftFormat)"
  exit 0
fi

PROJECT_ROOT=$(git rev-parse --show-toplevel)

for file in `git diff --cached --name-only --diff-filter=ACMR | grep -e '\(.*\).swift$'`
do
  swiftformat --config "${PROJECT_ROOT}/.swiftformat" "${file}"
  git add "$file";
done
```

I have seen the following [issue #374 ](https://github.com/nicklockwood/SwiftFormat/issues/374)
I have tried: 
`swiftformat stdin --stdinpath '{}' --config "${PROJECT_ROOT}/.swiftformat" "${file}"`
But then the formatting is not working at all.

The githook works perfectly otherwise and I would really prefer not to have a dependency on [git-format-staged](https://github.com/hallettj/git-format-staged). 
I know thats what documented in the README.

**Comment by nicklockwood** [comment]
The `--stdinpath '{}'` syntax is specific to git-format-staged and is only needed when formatting content from stdin instead of passing the actual file path.

I'm not sure why --exclude isn't working in your case since you are passing the file, but my guess is that it's because the relative path to the excluded files in your .swiftformat file isn't actually correct for the files being formatted?

What happens if you omit the --config parameter? Does it still pick up the .swiftformat file? It should do if the file is in the project root.

**Comment by martinmose** [comment]
Thanks for the quick reply!

That make sense.

I actually added the --config because I though the issue were that it couldn't find the `.swiftformat` file - but that isn't the case. It looks like it still pick up the file if i omit the --config parameter.

The swiftformat file look like this:
```
# format options
--commas inline
--maxwidth 150

--stripunusedargs closure-only

--wraparguments after-first
--wrapparameters after-first
--wrapcollections before-first

--trimwhitespace nonblank-lines

# file options
--exclude Project/Resources/Generated, MailCheck

# rules
--disable andOperator, markTypes, organizeDeclarations, numberFormatting, preferKeyPath, todos, wrapMultilineStatementBraces, wrapSwitchCases, enumNamespaces, redundantRawValues
```

I have also tried `--exclude **/Generated` it behavior is the same.

**Comment by martinmose** [comment]
@nicklockwood 
Just to follow up I have reproduced the issue with this command:
`swiftformat Project/Resources/Generated/Strings.swift`

Output:
```
Running SwiftFormat...
Reading config file at /root/.swiftformat
SwiftFormat completed in 0.32s.
1/1 files formatted.
```
As you can see it ignores the exclude option specified in the .swiftformat config.

I hope this make it easier to investigate when all the other things like githooks etc. is out of the way.

**Comment by shahzadmajeed** [comment]
I had same problem and following worked

In `.swiftformat` file

```
--exclude **/Pods/**.*
--exclude **/UnitTests/**.swift
--exclude **/TestApp/**.swift
```

**Comment by nicklockwood** [comment]
I believe the issue is that exclude rules that match a parent folder of the file being formatted don't work, but rules that match the file directly (including with a wildcard, as in @shahzadmajeed's example) do work. If so, I believe this is now fixed (or rather will be in the next release).

**Comment by nicklockwood** [comment]
@martinmose fixed in 0.49.5

**Comment by martinmose** [comment]
@nicklockwood Awesome! Thanks and it works :tada::tada:
